### PR TITLE
feat: add mtls for keymanager api requests

### PR DESCRIPTION
**Pull request by dracarys18**
## Type of Change

- [x] Enhancement

## Description

Add mTLS while calling keymanager service

### Additional Changes
- [x] This PR modifies application configuration/environment variables

## Motivation and Context

This PR adds a new `reqwest::Client`  with Certificates and CA to contact with the key manager

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code

### CODE REVIEW

File: crates/router/src/encryption.rs [hunk 1]
```diff
@@ -1,38 +1,116 @@
...
+
+        #[cfg(feature = "keymanager_mtls")]
+        {
+            let ca = state.conf.key_manager.ca.clone();
+            let cert = state.conf.key_manager.cert.clone();
+
+            let identity = reqwest::Identity::from_pem(cert.peek().as_ref())
+                .change_context(errors::ApiClientError::ClientConstructionFailed)?;
+            let ca_cert = reqwest::Certificate::from_pem(ca.peek().as_ref())
+                .change_context(errors::ApiClientError::ClientConstructionFailed)?;
```
Comment (Narayanbhat166) [inline crates/router/src/encryption.rs:34]: you can use 
```rust
if cfg!(feature = "keymanager_mtls") 
```

you would not need #[allow(unused_mut)] in that case

Comment (dracarys18) [reply crates/router/src/encryption.rs:34]: We cannot do this since ca and cert have compile time feature constraints

**Review by ArjunKarthik — APPROVED** [review]
LGTM

File: crates/router/src/encryption.rs [hunk 2]
```diff
@@ -1,38 +1,116 @@
...
 
 pub async fn call_encryption_service<T>(
     state: &SessionState,
     endpoint: &str,
     request_body: T,
-) -> errors::CustomResult<Result<Response, Response>, errors::ApiClientError>
+) -> errors::CustomResult<reqwest::Response, errors::ApiClientError>
 where
-    T: masking::ErasedMaskSerialize + Send + Sync + 'static,
+    T: serde::Serialize + Send + Sync + 'static,
```
Comment (Narayanbhat166) [inline crates/router/src/encryption.rs:93]: why have we removed masked serialize?

**Pull request merged**